src/PowerupSubsystem.cpp

```diff
@@ -32,12 +33,24 @@ void PowerupSubsystem::update() {
     frame_counter_ = std::min(POWERUP_LOCK_FRAMES, frame_counter_ + 1);
 }
 
+void PowerupSubsystem::add_mound_location(const int x, int y, const int z) {
+    charcoal_locations[mound_index] = glm::vec3(x, y + 2.7, z);
+
+    mound_index++;
```

Comment:
it would be better to just randomly select a mound index instead of iterating through them I think. Reduce predictability.